### PR TITLE
Update project dependencies to latest available

### PR DESCRIPTION
**Pull request by jyemin**
JAVA-4710

Note: Updating Jupiter dependency caused a strange Gradle build failure, so leaving that one alone for now.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -43,13 +43,13 @@ buildscript {
...
-    nettyTcnativeBoringsslVersion = '2.0.48.Final'
+    jnrUnixsocketVersion = '0.38.17'
+    nettyVersion = '4.1.79.Final'
+    nettyTcnativeBoringsslVersion = '2.0.54.Final'
     snappyVersion = '1.1.8.4'
-    zstdVersion = '1.5.0-4'
+    zstdVersion = '1.5.2-3'
     mongoCryptVersion = '1.6.0-alpha0'
-    projectReactorVersion = 'Californium-SR23'
+    projectReactorVersion = '2020.0.22'
```
Comment (rozza) [inline build.gradle:52]: Is this a big update / breaking change. I recall last time we were hesitant to use the latest version.

Comment (rozza) [reply build.gradle:52]: Ack

**Review by rozza — CHANGES_REQUESTED** [review]
Can you rebase? A `@NotNull` snuck in for a `@NonNull` in master breaking the tests.

**Comment by jyemin** [comment]
Yes, done.  Sorry about that, I don't know how I missed it.

**Review by rozza — APPROVED** [review]
LGTM!

**Pull request merged**